LPAL-857 Create a runbook to cover workspace extension (#1000)
ministryofjustice/opg-lpa: docs/runbooks/extending_workspace_time/extending_workspace.md

# Extending a PR workspace in development account

> this is a simple process, which is done by extending the `ExpiresTTL` for a given workspace name via the WorkspaceCleanup Dynamo DB table.

Author: ```suggestion
This process is used to extend the lifespan of a workspace; typically, an instance of the Make application to be used for UAT. This is useful for extended testing of complex PRs.
This is a simple process, which is done by extending the `ExpiresTTL` for a given workspace name via the WorkspaceCleanup Dynamo DB table.
```

steps:

1. Convert your desired date and time for expiry into a unix Epoch. A tool to help with this is <https://www.epochconverter.com/>
2. Open the Development Account console for AWS, assuming the operator role will be fine for this.
3. Search for `DynamoDB` in the top search bar and hit `<enter>`.
4. Look for `Tables` on the left hand side and click `Explore Items`. there will be a list of tables to select.
5. in the `Find tables by name` search box type `WorkspaceCleanup` and click the radio button next to the table name.
6. Look for the workspace name of your environment. to look for this:

   - Either: On your PR CI build, check the`post_environment_domains` job; the value is in the last step `print deployment values` next to `Terraform workspace:`
   - Or: list out these locally using `aws-vault exec identity -- terraform workspace list` and ascertain one that matches your PR.

7. Click on the workspace name link.
8. For the Attribute name `ExpiresTTL` place the converted Epoch time from step 1 into the Value box.
9. Click `Save Changes`

> 10. You now have extended the period of time for the workspace.

Author: ```suggestion
10. You now have extended the lifespan (Time To Live) for the workspace.
```

## Limitations of approach

> it is important to understand this will *only* be in place until:

Author: ```suggestion
it is important to understand the workspace will *only* available until:
```

- the TTL expires by itself.
- Someone pushes a change to the PR, which will reset to the standard 24 hours.

The environment will have until the next cleanup cycle before destruction takes place - i.e. 6am / 6pm UTC daily.

> Following the above will put this back in place.

Author: ```suggestion
Following the instructions for editing the workspace Time To Live will put this back in place.
```

This is a simple process, which is done by extending the `ExpiresTTL` for a given workspace name via the WorkspaceCleanup Dynamo DB table.

steps:

1. Decide when you would like to extend the workspace's lifespan to.
2. Convert that date and time for expiry into a unix Epoch. A tool to help with this is <https://www.epochconverter.com/>
3. Open the Development Account console for AWS, assuming the operator role will be fine for this.
4. Search for `DynamoDB` in the top search bar and hit `<enter>`.
5. Look for `Tables` on the left hand side and click `Explore Items`. there will be a list of tables to select.
6. in the `Find tables by name` search box type `WorkspaceCleanup` and click the radio button next to the table name.
7. Look for the workspace name of your environment. to look for this:

   - Either: On your PR CI build, check the`post_environment_domains` job; the value is in the last step `print deployment values` next to `Terraform workspace:`
   - Or: list out these locally using `aws-vault exec identity -- terraform workspace list` and ascertain one that matches your PR.

8. Click on the workspace name link.

> 9. For the Attribute name `ExpiresTTL` place the converted Epoch time from step 1 into the Value box.

Author: ```suggestion
9. For the Attribute name `ExpiresTTL` place the converted Epoch time from step 2 into the Value box.
```
